modules/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java

```diff
@@ -173,6 +208,29 @@ StorageOptions createStorageOptions(
         return new GoogleCloudStorageBlobContainer(randomBoolean() ? BlobPath.EMPTY : BlobPath.EMPTY.add("foo"), blobStore);
     }
 
+    public void testShouldRetryOnNetworkOutage() {
+        final int maxRetries = randomIntBetween(3, 5);
+
+        if (randomBoolean()) {
```

Comment:
Could we have two test cases, one for each branch, rather than picking randomly like this? They can both call the same underlying implementation.

Reply:
Addressed in 61598814212adcd09d06cc4cf3053077c66aa415